src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/RubyGemsUtils.cs

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.Sbom.Api.Exceptions;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Common;
+using Serilog;
+using File = System.IO.File;
+
+namespace Microsoft.Sbom.Api.PackageDetails;
+
+/// <summary>
+/// Utilities for retrieving information from rubygems packages that may not be present on the buildDropPath.
+/// </summary>
+public class RubyGemsUtils : IPackageManagerUtils<RubyGemsUtils>
+{
+    private readonly IFileSystemUtils fileSystemUtils;
+    private readonly ILogger log;
+    private readonly IRecorder recorder;
+
+    private readonly string rubyGemsPath;
+
+    public RubyGemsUtils(IFileSystemUtils fileSystemUtils, ILogger log, IRecorder recorder)
+    {
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+
+        this.rubyGemsPath = GetRubyGemsSpecificationsPath();
+    }
+
+    // Takes in a scanned component and attempts to find the associated gemspec file. If it is not found then it returns null.
+    public string? GetMetadataLocation(ScannedComponent scannedComponent)
```

Comment:
Having a hard time coming up with a way to unit test this since it relies on finding the gem executable which may not be present in test environments.